power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -104,6 +225,12 @@ template <dataset_type_tag dataset_type_> class Dataset {
     Buffer const& get_buffer(std::string_view component) const { return get_buffer(find_component(component, true)); }
     Buffer const& get_buffer(Idx i) const { return buffers_[i]; }
 
+    bool is_columnar(std::string_view component) const {
+        return buffers_[find_component(component, true)].data == nullptr;
+    }
+    bool is_columnar(Idx const i) const { return buffers_[i].data == nullptr; }
```

Comment:
try to keep the logic in one location
```suggestion
    constexpr bool is_columnar(Idx const i) const { return is_columnar(buffers_[i]); }
```